### DISCUSSION TITLE
Missing Architecture and Design Chapter?

### DISCUSSION THREAD

**Discussion by ehorley** [becarpenter/book6#59] [Ideas]
I know the focus is on operational deployment, but I would think there should be at a minimum some architecture and design section to talk about enterprise deployment compared to service providers or small businesses. My recommendation would be between Security and Deployment. Just my two cents.
Also, we typically do a lot of planning (IPv6 adoption plans), IPv6 assessments (to determine what is or is not going to work as part of the adoption) along with a proof of concept lab to validate designs (see above). Those are all missing - do they need to be included? Are they sections you think fit in other chapters and I am not seeing them?
- Ed

**Comment by becarpenter** [comment]
This is a very good point. I guess I'd like to think it through and see if anyone else comments, but I do think it would be better to collect this topic together rather than leave it scattered. Thanks!

**Reply by buraglio** [reply]
I lean toward having it in the flow of the book. All operational deployments have to start with a foundation, and the needs and requirements based architecture and design should be included as that foundation. If we are building a book to teach and aid current and future IPv6 practitioners, then we should assume that there will be a reasonable contingent that needs to start from zero, and a few reference architectures with reasoning will greatly aid in that.

**Comment by becarpenter** [comment]
Yes, Nick, but I don't think we can assume that people read the book in any particular order, so if someone looks at the Contents list it is probably a good idea for network design to be listed there. I'm inclined to agree with Ed that it fits best after Security. I'm also wondering whether it would be better to move the Deployment chapter, so we'd end up with
1. Introduction
2. IPv6 Basic Technology
3. Coexistence with Legacy IPv4
4. Security
5. Network Design
6. Management and Operations
7. Deployment Status
8. Case Studies
9. Further Reading

**Reply by ehorley** [reply]
I think this is a good idea, some thoughts... Should you make a new Chapter just called "Deployment" before Management and Operations? Typically we focus on an entire deployment strategy to reduce the risk and impacts of rolling out IPv6.
Then move into Case Studies and make Deployment Status just a section in a new Chapter called "Current Operational Practices" or something similar. I would be inclined to make it an appendix because Deployment Status is likely just pointing to third-party references (my guess.) I could be off the mark on this.

**Comment by becarpenter** [comment]
The current material in "Deployment" is really mainly about numbers of sites and users - so I think it really is "Deployment Status". I guess the main contributors there were looking it it a bit differently. What you're proposing is more "How to Plan a Deployment". 
I think that's important and I'm inclined to give it its own chapter. (There are no rules here.) Let's see your text and then decide.

I've made myself some notes on how to move and/or rename sections and chapters. It's not really hard, but I will have to write a tool to fix out-of-date internal cross references when a chapter or section is renamed. That's OK, just a rainy day programming project.

**Reply by buraglio** [reply]
I agree that having it in it's own chapter makes a lot of sense

**Comment by becarpenter** [comment]
I completed some testing on how to add/delete/rename chapters and sections. It doesn't seem worth writing a tool to do this, since it will always need some careful thought anyway, but I did write some code to look for internal citations broken by such actions (now included in indexBook.py) and fix some ugly code in makeBook.py. My notes on how to make such changes are a bit cryptic: https://github.com/becarpenter/book6/blob/main/utilities/chapterReorg.md

So, I think I'll rename the Deployment chapter as "Deployment Status" which is its real topic, and create an (empty) "Network Design" chapter, which is clearly needed.

(I don't attach much importance to the ordering of the chapters, because I think people will mainly dip into the book rather than reading in sequence. But I do want to keep the chapter titles short.)

Any further comments before I do that?

**Reply by ehorley** [reply]
No, that looks good to me, I will get current after that and start posting some chapter content this weekend.

**Comment by becarpenter** [comment]
OK... Chapter 5 is now an empty "Network Design" chapter. The previous Chapter 5 is now Chapter 8. Check https://github.com/becarpenter/book6/blob/main/Contents.md for details.
